### PR TITLE
CAMEL-17892: perform binding even when declarations were skipped

### PR DESCRIPTION
**Pull request by hanzo2001**
Force passive declaration before binding pre-existing queue and exchange.

Will only be applied when skipQueueBind=false and the other skip declarations are set to true

Issue tracker: https://issues.apache.org/jira/browse/CAMEL-17892

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
We should maybe look at camel-spring-rabbitmq too.

File: components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQDeclareSupport.java [hunk 1]
```diff
@@ -150,6 +155,18 @@ private void declareAndBindQueue(
         }
     }
 
+    private void passivelyDeclareExchangeAndQueueAndBindThem(
+            final Channel channel, final String queue, final String exchange, final String routingKey,
+            final Map<String, Object> bindingArgs)
+
+            throws IOException {
+
+        //if (!endpoint.isPassive()) { log.warn("Forcing passive because declaration was skipped"); }
```
Comment (davsclaus) [inline components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQDeclareSupport.java:164]: should this code comment be removed?

Comment (hanzo2001) [reply components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQDeclareSupport.java:164]: I didn't know what to do with a situation where the user may have defined passive=false but left the skip-declarations enabled.
I was thinking that logging (warning) this detail would be a good idea but I don't know if there is a logging policy (and the class does not even have its own logger).
I was hoping someone could enlighten me. I can clean it up either way

**Comment by hanzo2001** [comment]
@oscerd I am looking at `org.apache.camel.component.springrabbit.SpringRabbitMQEndpoint` but I do not see any mention to the documented skipping params:

- `camel.component.rabbitmq.skip-exchange-declarep`
- `camel.component.rabbitmq.skip-queue-declare`
- `camel.component.rabbitmq.skip-queue-bind`

To be honest, I have no experience working with Spring. I don't know if there is magic involved that I don't understand.

I can't find any routine in `SpringRabbitMQProducer` that is in charge of declaring anything.
`SpringRabbitMQConsumer` does reference `<Endpoint>::declareElements(...)`

```java
        if (getEndpoint().isAutoDeclare()) {
            // auto declare but without spring
            getEndpoint().declareElements(listenerContainer);
        }
```

However, after the routine has started, there is no mention to any skipping parameters. I don't understand much of the code so I am hoping someone more savvy with spring might help out.

**Comment by hanzo2001** [comment]
Can someone help me out with the build task error? How do I find the log that would show me what's wrong?

I think this has nothing to do with the component test because those are all green on my screen.

**Comment by orpiske** [comment]
> Can someone help me out with the build task error? How do I find the log that would show me what's wrong?
> 
> I think this has nothing to do with the component test because those are all green on my screen.

The log is available here: https://github.com/apache/camel/actions/runs/2077217542. Check down in the page for the "test-logs" artifact.

**Comment by hanzo2001** [comment]
@orpiske the logs do reference a "test-logs" artifact and a successful upload but there is no link to download them.

```
Failed components/camel-rabbitmq test test: 1 verified / 1 failed
Error: Process completed with exit code 1.
Run echo $PR_NUMBER > ./automated-build-log/pr_number
  echo $PR_NUMBER > ./automated-build-log/pr_number
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.14-1/x64
    PR_NUMBER: 7320
Run actions/upload-artifact@v3
  with:
    name: test-logs
    path: automated-build-log/**/*
    if-no-files-found: warn
  env:
    JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.14-1/x64
With the provided path, there will be 3 files uploaded
Starting artifact upload
For more detailed logs during the artifact upload process, enable step-debugging: https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging
Artifact name is valid!
Container for artifact "test-logs" successfully created. Starting upload of file(s)
Total size of all the files uploaded is 7703 bytes
File upload process has finished. Finalizing the artifact upload
Artifact has been finalized. All files have been successfully uploaded!

The raw size of all the files that were specified for upload is 146709 bytes
The size of all the files that were uploaded is 7703 bytes. This takes into account any gzip compression used to reduce the upload size, time and storage

Note: The size of downloaded zips can differ significantly from the reported size. For more information see: https://github.com/actions/upload-artifact#zipped-artifact-downloads 

Artifact test-logs has been successfully uploaded!
```

**Comment by orpiske** [comment]
> @orpiske the logs do reference a "test-logs" artifact and a successful upload but there is no link to download them.

It looks like Github only allow committers to download the artifacts. I attached it here. 

[test-logs.zip](https://github.com/apache/camel/files/8402119/test-logs.zip)

**Review by davsclaus — APPROVED** [review]

**Pull request merged**